api/routes/apds/activities/expenses/put.js

```diff
@@ -4,103 +4,92 @@ const {
   apdActivityExpense: defaultActivityExpenseModel,
   apdActivityExpenseEntry: defaultActivityExpenseEntryModel
 } = require('../../../../db').models;
-const { userCanEditAPD: defaultUserCanEditAPD } = require('../../utils');
-const loggedIn = require('../../../../auth/middleware').loggedIn;
+const {
+  loggedIn,
+  loadActivity,
+  userCanEditAPD
+} = require('../../../../middleware');
 
 module.exports = (
   app,
   ActivityModel = defaultActivityModel,
   ExpenseModel = defaultActivityExpenseModel,
-  ExpenseEntryModal = defaultActivityExpenseEntryModel,
-  userCanEditAPD = defaultUserCanEditAPD
+  ExpenseEntryModal = defaultActivityExpenseEntryModel
 ) => {
   logger.silly('setting up PUT /activities/:id/expenses route');
-  app.put('/activities/:id/expenses', loggedIn, async (req, res) => {
-    logger.silly(req, 'handling PUT /activities/:id/expenses route');
-    logger.silly(
-      req,
-      `attempting to update expenses on activity [${req.params.id}]`
-    );
-
-    try {
-      const activityID = +req.params.id;
-      const activity = await ActivityModel.where({ id: activityID }).fetch({
-        withRelated: ['apd', 'expenses.entries']
-      });
+  app.put(
+    '/activities/:id/expenses',
+    loggedIn,
+    loadActivity(),
+    userCanEditAPD(ActivityModel),
```

Comment:
Uses the activity model to find the activity with the ID given by `req.params.id`, then uses that object's relationships to get the APD.